README.md

```diff
@@ -1,136 +1,583 @@
+Install Playground from NPM:
 
-`my-element.ts`
+```sh
+npm install --save playground-elements
+```
 
-```ts
-import {LitElement, html, customElement} from 'lit-element';
+Import the `playground-elements` module to register the Playground custom
+elements:
 
-@customElement('my-element')
-class MyElement extends LitElement {
-  /* ... */
-}
+```html
+<script type="module">
+  import 'playground-elements';
+</script>
 ```
 
-`index.html`
+Create a `<playground-ide>` element in your HTML, and specify your project files
+inline:
 
 ```html
-<script type="module" src="my-element.js"></script>
+<playground-ide editable-filesystem line-numbers resizable>
+  <script type="sample/html" filename="index.html">
+    <!doctype html>
+    <body>
+      Hello
+      <script type="module" src="./index.js">&lt;/script>
+    </body>
+  </script>
+
+  <script type="sample/ts" filename="index.ts">
+    document.body.appendChild(document.createTextNode("World!"))
+  </script>
+</playground-ide>
+```
+
+Serve with any HTTP server that supports bare module import resolution, such
+as [@web/dev-server](https://modern-web.dev/docs/dev-server/overview/):
+
+```sh
+npm install --save-dev @web/dev-server
+npx web-dev-server --node-resolve --watch
 ```
 
-Note the filename of `my-element.js`.
+You can also use a tool like [Rollup](https://rollupjs.org/guide/en/) to resolve
+module imports to paths at build time. If you need more help with building and
+serving, check out the [Modern Web Guides](https://modern-web.dev/guides/).
 
-### Inline or external sources
+### Compatibility
 
-You can define an example entirely in HTML for simplicity:
+Playground is supported by all modern browsers. It requires support for custom
+elements, modules, service workers, and web workers.
+
+&nbsp;&nbsp;&nbsp; **Supported:** &nbsp;&nbsp;<img src="images/check-green.png" width="20px" height="20px" class="check" alt="Supported"> <img src="images/chrome.png" width="20px" height="20px"> Chrome
+&nbsp;&nbsp;&nbsp;
+<img src="images/check-green.png" width="20px" height="20px" class="check" alt="Supported"> <img src="images/firefox.png" width="20px" height="20px"> Firefox
+&nbsp;&nbsp;&nbsp;
+<img src="images/check-green.png" width="20px" height="20px" class="check" alt="Supported"> <img src="images/safari.png" width="20px" height="20px"> Safari
+&nbsp;&nbsp;&nbsp;
+<img src="images/check-green.png" width="20px" height="20px" class="check" alt="Supported"> <img src="images/edge.png" width="20px" height="20px"> Edge
+
+**Unsupported:** &nbsp; <img src="images/red-cross.png" width="20px" height="20px" class="check" alt="Unsupported"> <img src="images/ie.png" width="20px" height="20px"> Internet Explorer
+
+## Project files
+
+There are 3 ways to specify the files of a playground project.
+
+### Option 1: Inline scripts
+
+Add one or more `<script type="sample/..." filename="...">` tags as children of
```

Comment:
Can you document the supported types? Just a short list would do.

Reply:
Done.